Assets/Scripts/GameController.cs

```diff
@@ -25,6 +25,11 @@ public enum GameState { GAME_IN_PROGRESS, GAME_END }
 
     private float coinIncreaseTime;
 
+    public bool GetCurrentGameOver()
+    {
+        return currentGameState == GameState.GAME_END;
```

Comment:
I don't understand what you are trying to achieve here. SHouldn't it just return the current game state?